### ISSUE TITLE
Improve Selector usability (keyboard focus, hover and disabled states)

### ISSUE THREAD

**Issue by larimaza** [sumup-oss/circuit-ui#544]
## Component to amend

Selector

## Visual

<img width="202" alt="Captura de Tela 2020-02-07 às 11 21 10" src="https://user-images.githubusercontent.com/25252211/74036978-1d171800-499c-11ea-9032-eef1b22b721f.png">

## Context

We have a few issues to address on this:

- The Selector component is currently not accessible - not focusable via keyboard. That's because the markup structure is all `div`s. We could change them to radio buttons instead.
- Hover styles look kinda wonky because the thicker border pushes the elements around it.
- Disabled state does not look disabled enough - we'll need help from design team on this.
